server/server.go

```diff
@@ -603,15 +606,23 @@ func (s *Server) handlePublishWithoutResponse(r *http.Request, v *visitor) (*mes
 	if err != nil {
 		return nil, err
 	}
-	if !v.MessageAllowed() {
-		return nil, errHTTPTooManyRequestsLimitMessages
+	vRate := v
+	if topicCountsAgainst := t.Billee(); topicCountsAgainst != nil {
+		vRate = topicCountsAgainst
+	}
+
+	if !vRate.MessageAllowed() {
+		vRate = v
+		if !v.MessageAllowed() {
+			return nil, errHTTPTooManyRequestsLimitMessages
+		}
```

Comment:
If `v == vRate`, then this calls `v.MessageAllowed()` twice, no?

Reply:
Don't need that anyway if we're assuming a secret topic, and the subscriber taking the full load, so I got rid of the two stage process.